processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -1197,6 +909,42 @@ long getTimestamp()
     }
 
     @Override
+    public boolean equals(Object o)
+    {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+
+      TimeAndDims that = (TimeAndDims) o;
+
+      if (timestamp != that.timestamp) {
+        return false;
+      }
+      if (dims.length != that.dims.length) {
```

Comment:
Should this consider two TimeAndDims equal if their dims have different length, but the longer one is all nulls? That could happen if a dimension was added later on, but not all rows actually have that dimension. Those two rows should roll up together (I think IndexMerger actually will do this but maybe it'd be nice to do it here too).

I believe currently, the behavior you have here is equivalent to what TimeAndDimsComp does. So if we want to change that here then we should probably change it in the TimeAndDimsComp too.